### PR TITLE
Cleanup

### PR DESCRIPTION
**Pull request by tamird**
Some treasures from git stash.

### CODE REVIEW

File: kv/dist_sender.go [hunk 1]
```diff
@@ -503,21 +503,24 @@ func (ds *DistSender) sendAttempt(desc *proto.RangeDescriptor, call client.Call)
...
-	// replicas. In this case, likely all nodes are down (or not getting back
-	// to us within a reasonable amount of time).
-	// We may simply not be trying to talk to the up-to-date replicas, so
-	// clearing the descriptor here should be a good idea.
-	// TODO(tschottdorf): If a replica group goes dead, this will cause clients
-	// to put high read pressure on the first range, so there should be some
-	// rate limiting here.
-	if rpcErr != nil {
+	var err error
+	if rpcErr := ds.sendRPC(desc.RaftID, replicas, order, args, reply); rpcErr != nil {
```
Comment (bdarnell) [inline kv/dist_sender.go]: `rpcErr` is no longer needed as a separate variable if it is only used within this if. I'd spell this

    err := ds.sendRPC(...)
    if err != nil {
        // comments...
        ds.rangeCache.EvictCachedRangeDescriptor(...)
    } else {
        err = reply.Header().GoError()
    }

File: kv/dist_sender.go [hunk 2]
```diff
@@ -627,18 +630,20 @@ func (ds *DistSender) Send(_ context.Context, call client.Call) {
...
+		}(args.Header().Timestamp)
 		args.Header().Timestamp = ds.clock.Now()
 	}
 
 	// If this is a bounded request, we will change its bound as we receive
 	// replies. This undoes that when we return.
-	if args, ok := args.(proto.Bounded); ok && args.GetBound() > 0 {
+	boundedArgs, _ := args.(proto.Bounded)
+
+	if boundedArgs != nil {
```
Comment (bdarnell) [inline kv/dist_sender.go]: The check for `args.GetBound() > 0` got lost in this edit.

File: kv/dist_sender.go [hunk 3]
```diff
@@ -684,30 +689,29 @@ func (ds *DistSender) Send(_ context.Context, call client.Call) {
...
-			prevBound = args.GetBound()
-			if cReply, ok := curReply.(proto.Countable); ok {
-				if nextBound := prevBound - cReply.Count(); nextBound > 0 {
-					// Update bound for the next round.
-					// We've deferred restoring the original bound earlier.
-					args.SetBound(nextBound)
-				} else {
-					// Set flag to break the loop.
-					descNext = nil
+		if boundedArgs != nil {
```
Comment (bdarnell) [inline kv/dist_sender.go]: I'm not sure if it's an improvement to create a new boundedArgs variable whose lifetime spans 60+ lines of code. I think I'd rather leave this part as it was.

**Comment by bdarnell** [comment]
LGTM

File: kv/dist_sender.go (hunk 1 above)
Comment (tamird) [inline kv/dist_sender.go]: Nice. Done.

File: kv/dist_sender.go [hunk 4]
```diff
@@ -627,18 +628,20 @@ func (ds *DistSender) Send(_ context.Context, call client.Call) {
 	// required, set the timestamp using the local clock.
 	if args.Header().ReadConsistency == proto.INCONSISTENT && args.Header().Timestamp.Equal(proto.ZeroTimestamp) {
 		// Make sure that after the call, args hasn't changed.
-		defer func() {
-			args.Header().Timestamp = proto.ZeroTimestamp
-		}()
+		defer func(timestamp proto.Timestamp) {
```
Comment (tschottdorf) [inline kv/dist_sender.go]: This looks a lot less clear to me than it did before.

File: kv/dist_sender.go (hunk 2 above)
Comment (tschottdorf) [inline kv/dist_sender.go]: It's not necessary.

File: kv/dist_sender.go [hunk 5]
```diff
@@ -684,30 +687,29 @@ func (ds *DistSender) Send(_ context.Context, call client.Call) {
 		if finalReply != curReply {
 			// This was the second or later call in a multi-range request.
 			// Combine the new response with the existing one.
-			_, ok := finalReply.(proto.Combinable)
-			if !ok {
+			if cFinalReply, ok := finalReply.(proto.Combinable); ok {
+				cFinalReply.Combine(curReply)
+			} else {
```
Comment (tschottdorf) [inline kv/dist_sender.go]: Go guidelines suggest the
```
cond := ...
if cond {
   return
}
work()
```

pattern unless the scope gets polluted, which isn't the case here. I prefer the original.

File: kv/dist_sender.go (hunk 3 above)
Comment (tschottdorf) [inline kv/dist_sender.go]: ditto

**Comment by tschottdorf** [comment]
LGTM mod the parts with comments, which I think are noisy changes based on personal preference worth omitting.

**Pull request merged**